api/hasura/actions/createNominee.ts

```diff
@@ -0,0 +1,97 @@
+import assert from 'assert';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { z } from 'zod';
+
+import { getUserFromAddress } from '../../../api-lib/findUser';
+import {
+  insertNominee,
+  getUserFromProfileIdWithCircle,
+  getNomineeFromAddress,
+} from '../../../api-lib/nominees';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+import { GraphQLError } from '../../../src/lib/gql/zeusHasuraAdmin';
+import {
+  createNomineeSchemaInput,
+  composeHasuraActionRequestBody,
+} from '../../../src/lib/zod';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  try {
+    const {
+      input: { payload: input },
+      session_variables: sessionVariables,
+    } = composeHasuraActionRequestBody(createNomineeSchemaInput).parse(
+      req.body
+    );
+    if (sessionVariables.hasuraRole === 'admin') {
```

Comment:
You can use `composeHasuraActionRequestBodyWithSession` and it verifies user and not admin